src/monattack.cpp

```diff
@@ -561,6 +561,59 @@ void mattack::boomer(monster *z, int index)
     }
 }
 
+void mattack::boomer_glow(monster *z, int index)
+{
+    if( !z->can_act() ) {
+        return;
+    }
+
+    int t;
+    Creature *target = z->attack_target();
+    if( target == nullptr ||
+        rl_dist( z->pos(), target->pos() ) > 3 ||
+        !z->sees( *target, t ) ) {
+        return;
+    }
+
+    player *foe = dynamic_cast< player* >( target );
+    std::vector<tripoint> line = line_to( z->pos(), target->pos(), t, 0 );
+    z->reset_special(index); // Reset timer
+    z->moves -= 250;   // It takes a while
+    bool u_see = g->u.sees( *z );
+    if( u_see ) {
+        add_msg(m_warning, _("The %s spews bile!"), z->name().c_str());
+    }
+    for (auto &i : line) {
+        g->m.add_field(i, fd_bile, 1, 0);
+        if (g->m.move_cost(i) == 0) {
+            g->m.add_field(i, fd_bile, 3, 0);
+            if (g->u.sees( i ))
+                add_msg(_("Bile splatters on the %s!"), g->m.tername(i).c_str());
+            return;
+        }
+    }
+    if( !target->uncanny_dodge() ) {
+        if (rng(0, 10) > target->get_dodge() || one_in( target->get_dodge() ) ) {
+            target->add_env_effect("boomered", bp_eyes, 5, 25);
+            for (int i = 0; i < rng(2,4); i++){
+                body_part bp = random_body_part();
+                target->add_env_effect("glowing", bp, 4, 40);
+                if (target->has_effect("glowing")){
+                    break;
+                }
+            }
+        }
+        } else if( u_see ) {
+            target->add_msg_player_or_npc( _("You dodge it!"),
+                                           _("<npcname> dodges it!") );
+        }
+        if( foe != nullptr ) {
+            foe->practice( "dodge", 10 );
+            foe->ma_ondodge_effects();
```

Comment:
Now this can be cleaned up to `target->on_dodge( z, 10 )`. This way the entire `foe` won't be needed.